tests/gnrc_sock_tcp/main.c

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (C) 2021   Simon Brummer <simon.brummer@posteo.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include "shell.h"
+#include "msg.h"
+#include "net/sock/tcp.h"
+
+#define MAIN_QUEUE_SIZE (8)
+#define SOCK_TCP_QUEUE_SIZE (1)
+#define BUFFER_SIZE (1024)
+
+static msg_t main_msg_queue[MAIN_QUEUE_SIZE];
+static sock_tcp_t socks[SOCK_TCP_QUEUE_SIZE];
+static sock_tcp_t *sock = socks;
+static sock_tcp_queue_t sock_queue;
+static char buffer[BUFFER_SIZE];
+
+void dump_args(int argc, char **argv)
+{
+    printf("%s: ", argv[0]);
+    printf("argc=%d", argc);
+    for (int i = 0; i < argc; ++i) {
+        printf(", argv[%d] = %s", i, argv[i]);
+    }
+    printf("\n");
+}
+
+void print_result(const char *name, int err)
+{
+    if (err) {
+        printf("%s: returns %s\n", name, strerror(err));
+    } else {
+        printf("%s: returns %d\n", name, err);
+    }
+}
+
+int sock_tcp_connect_cmd(int argc, char **argv)
+{
+    dump_args(argc, argv);
+
```

Comment:
```suggestion
    if (argc < 2) {
        printf("usage: %s <host> <port>\n", argv[0]);
        return -1;
    }
 ```

let's be a bit more user friendly. Right now if you just call `gnrc_tcp_open` expecting to get usage information, the test will just crash on you.

Reply:
Also when I try

```
> gnrc_tcp_open fe80::90a7:a6ff:fe4b:2e32 12345
gnrc_tcp_open: argc=3, argv[0] = gnrc_tcp_open, argv[1] = fe80::90a7:a6ff:fe4b:2e32, argv[2] = 12345
gnrc_tcp_open: returns -EAFNOSUPPORT

> gnrc_tcp_open fe80::90a7:a6ff:fe4b:2e32%5 12345
gnrc_tcp_open: argc=3, argv[0] = gnrc_tcp_open, argv[1] = fe80::90a7:a6ff:fe4b:2e32%5, argv[2] = 12345
gnrc_tcp_open: returns -EAFNOSUPPORT
```

Reply:
I am not sure if user-friendlyness should be is the design goal for a test API only called from python code. 

For the -EAFNOSUPPORT thing I need to take a deeper look. This test should only expose sock_connect. Did you run this agains the gnrc_tcp tests instead of the gnrc_sock_tests that are the subject of this PR?

Reply:
Oh sorry I ran `tests/gnrc_tcp` instead of `tests/gnrc_sock_tcp` :sweat_smile: 

I also figured out the format

```
> sock_tcp_connect [fe80::90a7:a6ff:fe4b:2e32]:12345 6666
sock_tcp_connect: argc=3, argv[0] = sock_tcp_connect, argv[1] = [fe80::90a7:a6ff:fe4b:2e32]:12345, argv[2] = 6666
sock_tcp_connect: returns Unknown error -110
```

I wanted to run this against netcat since I expected it to behave very much like netcat, just with more manual steps

Reply:
Ah I have to specify the interface as well

```
> sock_tcp_connect [fe80::90a7:a6ff:fe4b:2e32%5]:12345 7412
sock_tcp_connect: argc=3, argv[0] = sock_tcp_connect, argv[1] = [fe80::90a7:a6ff:fe4b:2e32%5]:12345, argv[2] = 7412
sock_tcp_connect: returns 0
```

Have you considered using `netutils_get_ipv6()` in `gnrc_tcp_ep_from_str()` :wink:

Reply:
Hmm since we use strerror, most of the used errorcodes seem to be Unknown :(

I have not considered netutils_get_ipv6(), but I would argue that this is not in scope of this PR. Cleaning this up should happen as a seperate change. This PR is for the interface between gnrc and sock, not changes in gnrc.

Reply:
> Hmm since we use strerror, most of the used errorcodes seem to be Unknown :(

That's because `strerror` expects positive error codes

With this

```patch
--- a/tests/gnrc_sock_tcp/main.c
+++ b/tests/gnrc_sock_tcp/main.c
@@ -36,7 +36,7 @@ void dump_args(int argc, char **argv)
 void print_result(const char *name, int err)
 {
     if (err) {
-        printf("%s: returns %s\n", name, strerror(err));
+        printf("%s: returns %s\n", name, strerror(-err));
     } else {
         printf("%s: returns %d\n", name, err);
     }
```

I get
```
sock_tcp_read: returns Resource temporarily unavailable
```

instead of

```
sock_tcp_read: returns Unknown error -11
```

Reply:
I'll  fix it